src/Http/Controller/AdvsController.php

```diff
@@ -661,7 +661,11 @@ public function view($seo, $id = null)
             $adv = $this->adv_repository->getListItemAdv($id);
         }
 
-        if ($adv and ((auth()->user() and auth()->user()->hasRole('admin')) or ((!$adv->expired() && $adv->getStatus() === 'approved') || $adv->created_by_id === \auth()->id()))) {
+        if (!$adv){
```

Comment:
Let's use redirect whenever possible. Instead of pressing echo when there is no ad, let's redirect `abort(404);`.